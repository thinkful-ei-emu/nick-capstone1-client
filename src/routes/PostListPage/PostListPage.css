.post-list-filters {
  padding: 3% 0;
}


.filter-form{
  margin: 0 6%;
  padding: 3% 1%;
}

.filter-select {
  background-color: rgb(155, 163, 172);
  border: 1px solid rgb(102, 107, 112);
  font-weight: bold;
  border-radius: 8px;
  height: 36px;
  margin-left: 1%;
  min-height: 36px;
  color: white;
  font-size: 16px;
}

.filter-button {
  color: white;
  font-size: 16px;
  font-weight: bold;
  margin-left: 5px;
  min-height: 36px;
  width: auto;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(136, 33, 33);
  border-radius: 8px;
  margin-left: 20px;
}

.clear-control {
  padding-top: 2%;
}

.clear-button {
  color: white;
  font-size: 16px;
  font-weight: bold;
  min-height: 36px;
  width: auto;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(136, 33, 33);
  border-radius: 8px;
}

.create-post-button {
  color: white;
  font-size: 18px;
  font-weight: bold;
  min-height: 44px;
  width: auto;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(136, 33, 33);
  border-radius: 8px;
  margin-top: 3%;
  margin-bottom: 2%;
}


.post-list {
  border-top: 2px solid rgb(155, 163, 172);
  padding-top: 5px;
  padding-bottom: 10px;
  background-color: white;
}

@media only screen and (min-width: 500px){
  .post-list-filters {
    margin: 0 10%;
    padding-bottom: 0;
  }

  .filter-form{
    padding: 1% 1%;
  }
}


