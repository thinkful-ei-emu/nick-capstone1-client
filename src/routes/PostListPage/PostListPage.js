import React from 'react';
import { Link } from 'react-router-dom';
import PostListItem from '../../components/PostListItem/PostListItem';
import PostListContext from '../../contexts/PostListContext';
import PostApiService from '../../services/PostApiService';
import Lists from '../../Utils/Lists'

class PostListPage extends React.Component {

  static contextType = PostListContext;

  

  componentDidMount(){
    this.context.clearError()
    PostApiService.getAllPosts()
      .then(this.context.setPostList)
      .catch(this.context.setError)
  }
  
  handleFilter = ev => {
    ev.preventDefault()
    this.context.clearPostList()
    const {location_filter} = ev.target
    console.log(location_filter.value)
    PostApiService.getPostsByLocation(location_filter.value)
      .then(this.context.setPostList)
      .catch(this.context.setError)
  }

  handleClearFilter = ev => {
    ev.preventDefault();
    this.context.clearPostList()
    PostApiService.getAllPosts()
      .then(this.context.setPostList)
      .catch(this.context.setError)
  }


  renderPosts() {
    const { postList = []} =  this.context;
    return postList.map(post => 
      <PostListItem 
      key={post.id}
      post={post}/>
      )
  }

  render() {
    let locations = Lists.makeOptions(Lists.locationOptions);
    return (
      <div>
        <form onSubmit={this.handleFilter}>
          <select name='location_filter' id='location_filter'>
           {locations}
          </select>
          <button type='submit'>Filter Location</button>
          <button type='button' onClick={this.handleClearFilter}>Clear Filter</button>
        </form>
        <Link to={'/postform'}>Create Post</Link>
      {this.renderPosts()}
      </div>
    )
  }
}

export default PostListPage;