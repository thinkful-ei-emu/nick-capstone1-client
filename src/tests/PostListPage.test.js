import React from 'react';
import ReactDOM from 'react-dom';
import PostListPage from '../routes/PostListPage/PostListPage';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';



it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MemoryRouter><PostListPage /></MemoryRouter>, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><PostListPage /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
})