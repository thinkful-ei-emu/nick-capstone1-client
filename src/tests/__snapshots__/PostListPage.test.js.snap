// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
Array [
  <a
    href="/posts"
    onClick={[Function]}
  >
    <h1
      className="title-header"
    >
      BandBridge
    </h1>
  </a>,
  <header
    className="app-header"
  >
    <div
      className="header-login"
    >
      <h3
        className="user-name"
      >
        Guest
      </h3>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="login-button"
        >
          Login
        </button>
      </a>
        
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="login-button"
        >
          Sign up
        </button>
      </a>
    </div>
  </header>,
  <div
    className="post-list-filters"
  >
    <form
      className="filter-form"
      onSubmit={[Function]}
    >
      <select
        className="filter-select"
        defaultValue=""
        id="location_filter"
        name="location_filter"
      >
        <option
          disabled={true}
          value=""
        >
          Choose location
        </option>
        <option
          value="Montgomery (AL)"
        >
          Montgomery (AL)
        </option>
        <option
          value="Juneau (AK)"
        >
          Juneau (AK)
        </option>
        <option
          value="Phoenix (AZ)"
        >
          Phoenix (AZ)
        </option>
        <option
          value="Little Rock (AR)"
        >
          Little Rock (AR)
        </option>
        <option
          value="Sacramento (CA)"
        >
          Sacramento (CA)
        </option>
        <option
          value="Denver (CO)"
        >
          Denver (CO)
        </option>
        <option
          value="Hartford (CT)"
        >
          Hartford (CT)
        </option>
        <option
          value="Dover (DE)"
        >
          Dover (DE)
        </option>
        <option
          value="Tallahassee (FL)"
        >
          Tallahassee (FL)
        </option>
        <option
          value="Atlanta (GA)"
        >
          Atlanta (GA)
        </option>
        <option
          value="Honolulu (HI)"
        >
          Honolulu (HI)
        </option>
        <option
          value="Boise (ID)"
        >
          Boise (ID)
        </option>
        <option
          value="Springfield (IL)"
        >
          Springfield (IL)
        </option>
        <option
          value="Indianapolis (IN)"
        >
          Indianapolis (IN)
        </option>
        <option
          value="Des Moines (IA)"
        >
          Des Moines (IA)
        </option>
        <option
          value="Topeka (KS)"
        >
          Topeka (KS)
        </option>
        <option
          value="Frankfort (KY)"
        >
          Frankfort (KY)
        </option>
        <option
          value="Baton Rouge (LA)"
        >
          Baton Rouge (LA)
        </option>
        <option
          value="Augusta (ME)"
        >
          Augusta (ME)
        </option>
        <option
          value="Annapolis (MD)"
        >
          Annapolis (MD)
        </option>
        <option
          value="Boston (MA)"
        >
          Boston (MA)
        </option>
        <option
          value="Lansing (MI)"
        >
          Lansing (MI)
        </option>
        <option
          value="Saint Paul (MN)"
        >
          Saint Paul (MN)
        </option>
        <option
          value="Jackson (MS)"
        >
          Jackson (MS)
        </option>
        <option
          value="Jefferson City (MO)"
        >
          Jefferson City (MO)
        </option>
        <option
          value="Helena (MT)"
        >
          Helena (MT)
        </option>
        <option
          value="Lincoln (NE)"
        >
          Lincoln (NE)
        </option>
        <option
          value="Carson City (NV)"
        >
          Carson City (NV)
        </option>
        <option
          value="Concord (NH)"
        >
          Concord (NH)
        </option>
        <option
          value="Trenton (NJ)"
        >
          Trenton (NJ)
        </option>
        <option
          value="Santa Fe (NM)"
        >
          Santa Fe (NM)
        </option>
        <option
          value="Albany (NY)"
        >
          Albany (NY)
        </option>
        <option
          value="Raleigh (NC)"
        >
          Raleigh (NC)
        </option>
        <option
          value="Bismarck (ND)"
        >
          Bismarck (ND)
        </option>
        <option
          value="Columbus (OH)"
        >
          Columbus (OH)
        </option>
        <option
          value="Oklahoma City (OK)"
        >
          Oklahoma City (OK)
        </option>
        <option
          value="Salem (OR)"
        >
          Salem (OR)
        </option>
        <option
          value="Harrisburg (PA)"
        >
          Harrisburg (PA)
        </option>
        <option
          value="Providence (RI)"
        >
          Providence (RI)
        </option>
        <option
          value="Columbia (SC)"
        >
          Columbia (SC)
        </option>
        <option
          value="Pierre (SD)"
        >
          Pierre (SD)
        </option>
        <option
          value="Nashville (TN)"
        >
          Nashville (TN)
        </option>
        <option
          value="Austin (TX)"
        >
          Austin (TX)
        </option>
        <option
          value="Salt Lake City (UT)"
        >
          Salt Lake City (UT)
        </option>
        <option
          value="Montpelier (VT)"
        >
          Montpelier (VT)
        </option>
        <option
          value="Richmond (VA)"
        >
          Richmond (VA)
        </option>
        <option
          value="Olympia (WA)"
        >
          Olympia (WA)
        </option>
        <option
          value="Charleston (WV)"
        >
          Charleston (WV)
        </option>
        <option
          value="Madison (WI)"
        >
          Madison (WI)
        </option>
        <option
          value="Cheyenne (WY)"
        >
          Cheyenne (WY)
        </option>
      </select>
      <select
        className="filter-select"
        defaultValue=""
        id="instrument_filter"
        name="instrument_filter"
      >
        <option
          disabled={true}
          value=""
        >
          Choose instrument
        </option>
        <option
          value="Guitar(6)"
        >
          Guitar(6)
        </option>
        <option
          value="Guitar(7)"
        >
          Guitar(7)
        </option>
        <option
          value="Guiter(8)"
        >
          Guiter(8)
        </option>
        <option
          value="Acoustic-Guitar"
        >
          Acoustic-Guitar
        </option>
        <option
          value="Bass(4)"
        >
          Bass(4)
        </option>
        <option
          value="Bass(5)"
        >
          Bass(5)
        </option>
        <option
          value="Acoustic-Bass"
        >
          Acoustic-Bass
        </option>
        <option
          value="Double-Bass"
        >
          Double-Bass
        </option>
        <option
          value="Drums"
        >
          Drums
        </option>
        <option
          value="Vocals"
        >
          Vocals
        </option>
        <option
          value="Piano"
        >
          Piano
        </option>
        <option
          value="Keyboard"
        >
          Keyboard
        </option>
        <option
          value="Violin"
        >
          Violin
        </option>
        <option
          value="Saxophone"
        >
          Saxophone
        </option>
        <option
          value="Cello"
        >
          Cello
        </option>
        <option
          value="Flute"
        >
          Flute
        </option>
      </select>
      <button
        className="filter-button"
        type="submit"
      >
        Filter
      </button>
    </form>
    <div
      className="clear-control"
    >
      <button
        className="clear-button"
        onClick={[Function]}
        type="button"
      >
        Clear Filters
      </button>
    </div>
    <a
      href="/postform"
      onClick={[Function]}
    >
      <button
        className="create-post-button"
      >
        Create Post
      </button>
    </a>
  </div>,
  <div
    className="post-list"
  />,
]
`;
